### ISSUE TITLE
Running the tests on windows

### ISSUE THREAD

**Issue by SylvainCorlay** [SRombauts/SQLiteCpp#280]
Hi, many thanks for your work on this package. We use it for the [xeus-sqlite](https://github.com/jupyter-xeus/xeus-sqlite) project, a Jupyter kernel for SQLite, which you can try by clicking [here](https://mybinder.org/v2/gh/jupyter-xeus/xeus-sqlite/stable).

I am working on packaging SQLiteCpp on conda-forge for Windows (OS X and Linux builds are already availble).

- Our build at https://github.com/conda-forge/sqlitecpp-feedstock/pull/1 is succesful (https://github.com/conda-forge/sqlitecpp-feedstock/pull/1/commits/490b6c21a011178a890c1c3bba2c67068c710a71)
- But enabling the tests (https://github.com/conda-forge/sqlitecpp-feedstock/pull/1/commits/2ab08b0c5a34c078114e08dd98fce68e4eb25161) results in the following [failure](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=163139&view=logs&j=f97ca392-d626-52d0-4b2e-4b27aa319a77):

<details>

```
2020-05-16T09:50:21.4855913Z LINK: command "C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\SQLiteCpp_tests.dir\objects1.rsp /out:SQLiteCpp_tests.exe /implib:SQLiteCpp_tests.lib /pdb:%SRC_DIR%\build\SQLiteCpp_tests.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console SQLiteCpp.lib %PREFIX%\Library\lib\gtest-md.lib %PREFIX%\Library\lib\gtest_main-md.lib %PREFIX%\Library\lib\sqlite3.lib %PREFIX%\Library\lib\gtest-md.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:SQLiteCpp_tests.exe.manifest" failed (exit code 1120) with the following output:
2020-05-16T09:50:21.4863167Z gtest-md.lib(gtest-all.cc.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2020-05-16T09:50:21.4866249Z gtest-md.lib(gtest-all.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in Column_test.cpp.obj
2020-05-16T09:50:21.4882165Z gtest_main-md.lib(gtest_main.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in Column_test.cpp.obj
2020-05-16T09:50:21.4883599Z LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library
2020-05-16T09:50:21.4884333Z Transaction_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OK" (?OK@SQLite@@3HB)
2020-05-16T09:50:21.4886138Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OK" (?OK@SQLite@@3HB)
2020-05-16T09:50:21.4887717Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OK" (?OK@SQLite@@3HB)
2020-05-16T09:50:21.4888380Z Statement_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OK" (?OK@SQLite@@3HB)
2020-05-16T09:50:21.4888972Z Backup_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OK" (?OK@SQLite@@3HB)
2020-05-16T09:50:21.4889567Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::INTEGER" (?INTEGER@SQLite@@3HB)
2020-05-16T09:50:21.4890145Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::FLOAT" (?FLOAT@SQLite@@3HB)
2020-05-16T09:50:21.4890735Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::TEXT" (?TEXT@SQLite@@3HB)
2020-05-16T09:50:21.4892014Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::BLOB" (?BLOB@SQLite@@3HB)
2020-05-16T09:50:21.4892646Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::Null" (?Null@SQLite@@3HB)
2020-05-16T09:50:21.4893246Z Transaction_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4893916Z VariadicBind_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4894563Z ExecuteMany_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4895217Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4895883Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4896578Z Statement_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4897361Z Backup_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READWRITE" (?OPEN_READWRITE@SQLite@@3HB)
2020-05-16T09:50:21.4898015Z Transaction_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4898667Z VariadicBind_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4899292Z ExecuteMany_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4899976Z Column_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4900620Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4901266Z Statement_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4901877Z Backup_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_CREATE" (?OPEN_CREATE@SQLite@@3HB)
2020-05-16T09:50:21.4902515Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READONLY" (?OPEN_READONLY@SQLite@@3HB)
2020-05-16T09:50:21.4903142Z Backup_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::OPEN_READONLY" (?OPEN_READONLY@SQLite@@3HB)
2020-05-16T09:50:21.4903845Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "char const * const SQLite::VERSION" (?VERSION@SQLite@@3PEBDEB)
2020-05-16T09:50:21.4904505Z Database_test.cpp.obj : error LNK2001: unresolved external symbol "int const SQLite::VERSION_NUMBER" (?VERSION_NUMBER@SQLite@@3HB)
2020-05-16T09:50:21.4905180Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__ftime64
2020-05-16T09:50:21.4905636Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__localtime64_s
2020-05-16T09:50:21.4906102Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_fdopen
2020-05-16T09:50:21.4906527Z OLDNAMES.lib(fdopen.obi) : error LNK2001: unresolved external symbol __imp_fdopen
2020-05-16T09:50:21.4906972Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_dup2
2020-05-16T09:50:21.4907424Z OLDNAMES.lib(dup2.obi) : error LNK2001: unresolved external symbol __imp_dup2
2020-05-16T09:50:21.4907894Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__strdup
2020-05-16T09:50:21.4908331Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_creat
2020-05-16T09:50:21.4908815Z OLDNAMES.lib(creat.obi) : error LNK2001: unresolved external symbol __imp_creat
2020-05-16T09:50:21.4909258Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_read
2020-05-16T09:50:21.4909683Z OLDNAMES.lib(read.obi) : error LNK2001: unresolved external symbol __imp_read
2020-05-16T09:50:21.4910116Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_dup
2020-05-16T09:50:21.4910526Z OLDNAMES.lib(dup.obi) : error LNK2001: unresolved external symbol __imp_dup
2020-05-16T09:50:21.4910975Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__stat64i32
2020-05-16T09:50:21.4911462Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__getcwd
2020-05-16T09:50:21.4911942Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_wcsstr
2020-05-16T09:50:21.4912379Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__stricmp
2020-05-16T09:50:21.4912844Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_strstr
2020-05-16T09:50:21.4913335Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp__mkdir
2020-05-16T09:50:21.4913792Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_strerror
2020-05-16T09:50:21.4914224Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_write
2020-05-16T09:50:21.4914665Z OLDNAMES.lib(write.obi) : error LNK2001: unresolved external symbol __imp_write
2020-05-16T09:50:21.4915118Z gtest-md.lib(gtest-all.cc.obj) : error LNK2001: unresolved external symbol __imp_close
2020-05-16T09:50:21.4915587Z OLDNAMES.lib(close.obi) : error LNK2001: unresolved external symbol __imp_close
2020-05-16T09:50:21.4916002Z OLDNAMES.lib(fdopen.obi) : error LNK2001: unresolved external symbol __imp__fdopen
2020-05-16T09:50:21.4916415Z OLDNAMES.lib(dup2.obi) : error LNK2001: unresolved external symbol __imp__dup2
2020-05-16T09:50:21.4916851Z OLDNAMES.lib(creat.obi) : error LNK2001: unresolved external symbol __imp__creat
2020-05-16T09:50:21.4917261Z OLDNAMES.lib(dup.obi) : error LNK2001: unresolved external symbol __imp__dup
2020-05-16T09:50:21.4917669Z SQLiteCpp_tests.exe : fatal error LNK1120: 32 unresolved externals
2020-05-16T09:50:21.4937040Z NMAKE : fatal error U1077: 'D:\bld\sqlitecpp_1589622461776\_build_env\Library\bin\cmake.exe' : return code '0xffffffff'
2020-05-16T09:50:21.4937605Z Stop.
2020-05-16T09:50:21.4955024Z NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
2020-05-16T09:50:21.4955902Z Stop.
2020-05-16T09:50:21.4969824Z NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
2020-05-16T09:50:21.4970476Z Stop.
```

</details>

**Comment by SRombauts** [comment]
Hello,

I think that your link-time error is related to MSVC runtime library incompatibility between your application and the unit test library I use (Google test).
I am using it by specifying explicitly the runtime to be used, that is not the default one.

Could you have a look at **#236 Disable explicit setting of MSVC runtime** that should solve this?

Let me know.

**Comment by SylvainCorlay** [comment]
Thanks for your reply @SRombauts.

To clarify, I am not building a specific application, just the tests of SQLiteCpp, as part of the packaging of sqlitecpp for conda.

I will look into the linked PR.

**Comment by SylvainCorlay** [comment]
Hey @SRombauts, thanks for the pointer (and it's nice to see @emmenlau here).

I am not sure that we should change the default on /MT and /MD for the library depending on whether we build the tests or not.

**Comment by emmenlau** [comment]
Hey @SylvainCorlay , small world! So to confirm you would promote the PR #236 Disable explicit setting of MSVC runtime in order *not* to change the default on /MT and /MD for the library depending on whether we build the tests or not, correct?

@SRombauts , I know there are not so many advocates for PR #236 yet, but do you think we could give it a try to see if it works for the larger community?

**Comment by SylvainCorlay** [comment]
> Hey @SylvainCorlay , small world! So to confirm you would promote the PR #236 Disable explicit setting of MSVC runtime in order not to change the default on /MT and /MD for the library depending on whether we build the tests or not, correct?

Correct.

**Comment by emmenlau** [comment]
Just kindly promoting this issue (and matching PR) again...

**Comment by SRombauts** [comment]
The related PR have now both been merged
Disabling explicit setting of MSVC had been removed from the PR but I did made a similar fix yesterday in #370 I believe 

Cheers!
